sdk/cosmos/azure_data_cosmos/src/constants.rs

```diff
@@ -40,3 +40,219 @@ pub const SUB_STATUS: HeaderName = HeaderName::from_static("x-ms-substatus");
 pub const QUERY_CONTENT_TYPE: ContentType = ContentType::from_static("application/query+json");
 
 pub(crate) const PREFER_MINIMAL: HeaderValue = HeaderValue::from_static("return=minimal");
+
+pub const ACCOUNT_PROPERTIES_KEY: &str = "account_properties_key";
+
+#[repr(u16)]
```

Comment:
I'd avoid using `#[repr(u16)]` unless you're calling or exposing C APIs (which we're not doing... **yet**). You're already doing the right thing below, and using a big `match` to map integers to enum variants (rather than just casting the integer to the enum, like you might do in .NET). Rust conventions strongly discourage doing that, because the whole Rust ecosystem expects that an enum can **only** have one of the values listed in it's definition, and **no other value**.

Since we already have the "mapping" between `u16` and `SubStatusCode` in the `TryFrom<u16>` implementation, we can let Rust pick whatever in-memory representation makes sense based on the situation. That allows to compiler to optimize memory usage.

Reply:
Thanks for the suggestion here. I have removed the `#[repr(u16)]` usage.